test/box/gh-6634-different-log-on-tuple-new-and-free.test.lua

```diff
@@ -0,0 +1,31 @@
+env = require('test_run')
```

Comment:
Diff tests are now outlawed unfortunately. Suggest you to rewrite this test as tap, it shouldn't be complicated.

Reply:
Otherwise LGTM, this patch can be pushed out of order.

Reply:
The test uses `grep` on log with `log_level = 7` and diff test have `grep_log` function which provides this opportunity while tap test did not cope with it (With `log_level = 7` socket just has been broken on attempt write log to it).

Reply:
So try lua-test :)

Reply:
I guess nothing prevents a 'core = app' (so called tap) test from using `test_run:grep_log()`. At least using [explicit log file name](https://github.com/tarantool/test-run/issues/196).